vm/src/obj/objrange.rs

```diff
@@ -224,3 +233,20 @@ fn range_index(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
         unreachable!()
     }
 }
+
+fn range_count(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(
+        vm,
+        args,
+        required = [(zelf, Some(vm.ctx.range_type())), (needle, None)]
+    );
+
+    if let PyObjectPayload::Range { ref range } = zelf.borrow().payload {
+        match needle.borrow().payload {
```

Comment:
Instead of using the PyObjectPayload here, use the vm.isinstance type check to determine int type, and objint::get_value to retrieve the BigInt value.

Reply:
This works for the argument, but won't I still eventually need to use `PyObjectPayload` to pull out the actual `RangeType`? Apologies if I misunderstood you, I'm not entirely familiar with this project.

Reply:
No worries, thank you for your help! The idea is to use PyObjectPayload as little as possible. The objrange.rs file should contain a method get_value which will retrieve the payload for you. This saves a lot of duplicate code for borrow and matching on the payload type. Take a look at the objint::getvalue method for example. It returns you the BigInt contents of the integer. The same should be there for the range type.

Hope this helps!

Reply:
Btw, the problem with checking strictly on the payload is that this will eventually fail with subclasses. The isinstance check is more thorough and works for subclasses as well!